src/libcrun/container.c

```diff
@@ -1120,6 +1135,16 @@ container_init_setup (void *args, pid_t own_pid, char *notify_socket, int sync_s
   if (UNLIKELY (ret < 0))
     return ret;
 
+#if HAVE_DLOPEN && HAVE_LIBKRUN
+  /* explicitly configure kvm device if binary is invoked as krun */
+  if (entrypoint_args->context->handler != NULL && (strcmp (entrypoint_args->context->handler, "krun") == 0))
+    {
+      ret = libcrun_create_kvm_device (container, err);
+      if (UNLIKELY (ret < 0))
+        return ret;
+    }
+#endif
```

Comment:
@flouthoc @slp does the `libcrun_create_kvm_device` need to happen also when the handler is specified through an annotation?

Reply:
@giuseppe No but then it is expected that `/dev/kvm` is configured from OCI configuration.

Reply:
@giuseppe @flouthoc Actually, I think it'd be nice if `crun` automatically made `/dev/kvm` accessible for all cases that lead to `libkrun` being used.

Reply:
@slp Sure we can also add `kvm` by default for annotation based handler as well.  @giuseppe WDYT ?

Reply:
yes, I think it is a good idea and that will also simplify some of the new code

Reply:
cool I'll modify code for this.